### PR TITLE
Update run state documentation

### PR DESCRIPTION
**Pull request by tomasmik**
# Description of the change

## Checklist

Please make sure that the proposed change checks all the boxes below before requesting a review:

- [x] I have reviewed the [guidelines for contributing](https://github.com/spacelift-io/user-documentation/blob/main/CONTRIBUTING.md) to this repository.
- [x] The preview looks fine.
- [x] The tests pass.
- [x] The commit history is clean and meaningful.
- [x] The pull request is opened against the `main` branch.
- [x] The pull request is no longer marked as a draft.
- [x] You agree to license your contribution under the [MIT license](https://github.com/spacelift-io/user-documentation/blob/main/LICENSE) to Spacelift (not required for Spacelift employees).
- [ ] You have updated the navigation files correctly:
    - [x] No new pages have been added, or;
    - [ ] Only _nav.yaml_ has been updated because the changes only apply to SaaS, or;
    - [ ] Only _nav.self-hosted.yaml_ has been updated because the changes only apply to Self-Hosted, or;
    - [ ] Both _nav.yaml_ and _nav.self-hosted.yaml_ have been updated.

If the proposed change is ready to be merged, please request a review from `@spacelift-io/solutions-engineering`. Someone will review and merge the pull request.

_Spacelift employees should request reviews from the relevant engineers and are allowed to merge pull requests after they got at least one approval._

Thank you for your contribution! 🙇

### CODE REVIEW

File: docs/concepts/run/README.md [hunk 1]
```diff
@@ -25,21 +25,34 @@ Regardless of the type of the job performed, some phases and terminal states are
...
+- It's waiting for external dependencies to finish
+
+ Queued is a _passive state_ meaning no operations are performed while a run is in this state. The user can also cancel the run while it's still queued, transitioning it to the terminal [Canceled](#canceled) state.
+
+### Ready
+
+Ready state means that the run is eligible for processing and is waiting for a worker to become available. A run will stay in this state until
+it's picked up by a worker.
+
+If you're using the public worker pool, you can track its availability on our [status page](https://spacelift.statuspage.io/){: rel="nofollow"}. In particular, you should look at a system metric called _Public worker queuing time._ It indicates how long has the oldest run spent in the queue since becoming eligible for processing:
```
Comment (tomasmik) [inline docs/concepts/run/README.md:49]: Should we remove this or are we keeping the graph just changing how it works? Not sure whats the latest on this.

File: docs/concepts/run/README.md [hunk 2]
```diff
@@ -25,21 +25,34 @@ Regardless of the type of the job performed, some phases and terminal states are
 
 ### Queued
 
-Queued means that **no worker has picked up the run yet**. There are two possible reasons for that: there are no available workers, or the run is not eligible for processing.
+Queued means that the run is not [Ready](#preparing) for processing, as it's either blocked, waiting for for dependencies to finish or requires additional action from a user.
```
Comment (mbialon) [inline docs/concepts/run/README.md:28]:
```suggestion
Queued means that the run is not [Ready](#ready) for processing, as it's either blocked, waiting for dependencies to finish, or requires additional action from a user.
```

File: docs/concepts/run/README.md [hunk 3]
```diff
@@ -25,21 +25,34 @@ Regardless of the type of the job performed, some phases and terminal states are
...
 
-If you run or task is **currently blocked** by something else holding the lock on the stack, you'll see the link to the blocker in the header:
+If your run or task is **currently blocked** by something else holding the lock on the stack, you'll see the link to the blocker in run state list:
 
-![](../../assets/screenshots/01DTD3M6DVC7VKQGJ72PCSEBKD_·_End-to-end_testing.png)
+![](../../assets/screenshots/blocked-queued-run-unconfirmed.png)
 
-The other scenario is just **running out of available workers**. If you're using the public worker pool, you can track its availability on our [status page](https://spacelift.statuspage.io/){: rel="nofollow"}. In particular, you should look at a system metric called _Public worker queuing time._ It indicates how long has the oldest run spent in the queue since becoming eligible for processing:
+There can also be other reasons why the run is in this state and is not being promoted to state [Ready](#preparing):
```
Comment (mbialon) [inline docs/concepts/run/README.md:36]:
```suggestion
There can also be other reasons why the run is in this state and is not being promoted to state [Ready](#ready):
```

**Review by mbialon — DISMISSED** [review]

File: docs/concepts/run/README.md (hunk 3 above)
Comment (tomasmik) [reply docs/concepts/run/README.md:36]: oops, didn't notice i copied these wrong. good catch.

**Review by jmfontaine — COMMENTED** [review]
I agree with @mbialon's comments. I do not have anything to add.

File: docs/concepts/run/README.md (hunk 1 above)
Comment (cube2222) [reply docs/concepts/run/README.md:49]: Remove

**Review by cube2222 — DISMISSED** [review]

File: docs/concepts/run/README.md [hunk 4]
```diff
@@ -25,21 +25,31 @@ Regardless of the type of the job performed, some phases and terminal states are
 
 ### Queued
 
-Queued means that **no worker has picked up the run yet**. There are two possible reasons for that: there are no available workers, or the run is not eligible for processing.
+Queued means that the run is not [Ready](#ready) for processing, as it's either blocked, waiting for for dependencies to finish or requires additional action from a user.
```
Comment (mbialon) [inline docs/concepts/run/README.md:28]: I think a comma is appropriate before the last *or*.

**Review by mbialon — APPROVED** [review]

**Pull request merged**